doc/Language/mop.pod6

```diff
@@ -222,6 +222,82 @@ the subtleties of Raku in order to avoid pitfalls when working with the MOP,
 and can't expect the same L<"do what I mean"|/language/glossary#DWIM>
 convenience that ordinary Raku code offers.
 
+X<|Archetypes>
+
+=head1 Archetypes
+
+Typically, when multiple kinds of types share a property, it is
+implemented with a metarole to be mixed into their metaclasses. However,
+not all common properties of types can be implemented as mixins. One
+example of this is parameterization; while roles are the only kind in
+Rakudo that support this (ordinarily), types requiring arguments in
+order to be complete is not a concept unique to roles. Properties like
+this are known as I<archetypes>.
+
+HOWs should provide a C<archetypes> metamethod that takes no arguments
+and returns a C<Metamodel::Archetypes> instance. This is used by the
+compiler to determine what archetypes are supported by a metaobject.
+The rest of this section will cover how each of the archetypes that
+exist in Rakudo work.
```

Comment:
I would suggest putting less focus on parameterization. It would be better to put more stress on archetypes describing what a type object _is_ and what it's purpose.

Reply:
Also, add an example.